src/main/java/uk/gov/companieshouse/api/accounts/interceptor/SmallFullInterceptor.java

```diff
@@ -0,0 +1,79 @@
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import uk.gov.companieshouse.api.accounts.AttributeName;
+import uk.gov.companieshouse.api.accounts.LinkType;
+import uk.gov.companieshouse.api.accounts.model.entity.CompanyAccountEntity;
+import uk.gov.companieshouse.api.accounts.model.entity.SmallFullEntity;
+import uk.gov.companieshouse.api.accounts.service.SmallFullService;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+/**
+ * The SmallFullInterceptor interceptor validates that the Smallfull entity is in the correct state
+ * for the calling request. It insures a Smallfull resource exists and matches the one stored in the
+ * CompanyAccount provided in the session. This interceptor will fail if the CompanyAccount
+ * interceptor has not run prior to its own execution.
+ */
+@Component
+public class SmallFullInterceptor extends HandlerInterceptorAdapter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("company-accounts.api.ch.gov.uk");
+    @Autowired
+    private SmallFullService smallFullService;
+
+    /**
+     * This class validates a Small-full account exists for the given CompanyAccount. Validation is
+     * carried out via a database lookup for the SmallFullEntity Object then matching the retrieved
+     * Entities 'self' link to the session stored CompanyAccountEntities 'small_full_accounts' link.
+     * Providing this validation passes it assigns the SmallFullEntity to the session.
+     *
+     * @param request - current HTTP request
+     * @param response - current HTTP response
+     * @param handler - chosen handler to execute, for type and/or instance evaluation
+     * @return true if the execution chain should proceed with the next interceptor or the handler
+     * itself.
+     */
+    @Override
+    @SuppressWarnings("unchecked")
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response,
+            Object handler) throws NoSuchAlgorithmException {
+
+        HttpSession session = request.getSession();
+        CompanyAccountEntity companyAccountEntity = (CompanyAccountEntity) session
+                .getAttribute(AttributeName.COMPANY_ACCOUNT.getValue());
+        if (companyAccountEntity == null) {
+            LOGGER.error(
+                    "SmallFullInterceptor failed on preHandle: Failed to retrieve a companyAccountEntity from the session.");
+            return false;
+        }
+
+        String companyAccountId = companyAccountEntity.getId();
+        String smallFullId = smallFullService.generateID(companyAccountId);
+        SmallFullEntity smallFull = smallFullService.findById(smallFullId);
+        if (smallFull == null) {
+            LOGGER.error(
+                    "SmallFullInterceptor failed on preHandle: Failed to retrieve a SmallFull account from the Database.");
+            return false;
```

Comment:
set response code before return false - 500 here if we catch a MongoException or 404 if null is returned?